graph/patterns/change-status.md

```diff
@@ -0,0 +1,86 @@
+# Change Status & Action Request
+
+Microsoft Graph API Design Pattern
+
+_The change status pattern is an effective alternative to model the ability to trigger side effects while keeping an agnostic, data driven approach_
+
+## Problem
+
+In some cases, exposed resources can trigger side effects that modify the system behavior. For instance, a [`riskyUser`](https://docs.microsoft.com/en-us/graph/api/resources/riskyuser?view=graph-rest-1.0) can be [`confirmed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http) or [`dismissed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http).
+
+These situation are often modeled as a `POST` request to a specific endpoint in the same URI space of the resource, triggering the side effect directly and not returning any data. While it might be a quick and easy solution, there are some considerations:
+
+- The API should be exposing resources and allow their manipulation, while Actions resemble an object in [OOP][1]
+- It couples the request with the side effect, making more difficult to test, log, replay and reason about what is happening in the system
+- It forces the consumer to follow the [OOP][1] approach at the API level. We do not want to pursue any specific style and the developers writing the client might be working in other ways
+
+## Solution
+
+API Designers can prefer the usage of a `status` property or a datafyed version of the intended action to represent the **intent** of triggering a side effect rather than starting the execution directly through the HTTP request.
+
+Taking in consideration the example of the `riskyUser` above, it is possible to add a `status` property to the resource:
+
+```xml
+<EnumType Name="riskyStatus">
+    <Member Name="confirmed" Value="0"/>
+    <Member Name="dismissed" Value="1"/>
+    <Member Name="unknown" Value="2"/>
+</EnumType>
+
+<EntityType Name="riskyUser" >
+    <Property Name="status" Type="riskyStatus" />
+</EntityType>
+```
+
+It is now possible to change the status through a PATCH request to the entity:
+
+```bash
+curl -X PATCH --json '{"status": "confirmed"}' https://graph.microsoft.com/v1.0/riskyUsers/a57dc75f-24b5-47ce-b5e1-44822f5d4729
+```
+
+Instead of returning an empty body with `204` status code, depending on the situation, the response should contain the status of the request, if can be satisfied in a reasonable time:
+
+```json
+{
+  "newStatus": "confirmed"
+}
+```
+
+If the response triggers a long running operation, the response should contain a reference to query its status, following the [existing guidance][2]. The status on the resource shall stay the same until the long running operation is completed:
```

Comment:
The link should be updated to the LRO topic that we should have on this site, rather than the internal site.

Reply:
Do we have such topic here? I cannot find any document in the repository. What document are you referring to?